### ISSUE TITLE
[Phase 3] Baseline Model

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/exoplanet-detection-ml#4]
# Phase 3: Baseline Model

**Parent Issue:** #1

## Objective

Train and evaluate Random Forest classifier for transit detection.

## Dependencies

- Phase 1 completed (data available)
- Phase 2 completed (features extracted)

### Model Implementation
- [ ] Implement `src/models/random_forest.py`
- [ ] Create `TransitClassifier` wrapper class
- [ ] Add hyperparameter configuration via YAML
- [ ] Implement model saving/loading

### Training Pipeline
- [ ] Implement `src/models/train.py`
- [ ] Handle class imbalance (SMOTE or class weights)
- [ ] Implement cross-validation (stratified k-fold)
- [ ] Add MLflow logging for experiment tracking

### Evaluation
- [ ] Implement `src/models/evaluate.py`
- [ ] Compute AUC, precision, recall, F1
- [ ] Generate confusion matrix
- [ ] Create ROC and precision-recall curves
- [ ] Analyze feature importance

### Hyperparameter Tuning
- [ ] Grid search over n_estimators, max_depth, min_samples_split
- [ ] Document best hyperparameters

### Model Wrapper
```python
class TransitClassifier:
    """Random Forest classifier for transit detection."""

    def __init__(self, n_estimators: int = 100, class_weight: str = 'balanced'):
        self.model = RandomForestClassifier(
            n_estimators=n_estimators,
            class_weight=class_weight,
            n_jobs=-1,
            random_state=42
        )

    def fit(self, X: np.ndarray, y: np.ndarray) -> 'TransitClassifier':
        self.model.fit(X, y)
        return self

    def predict_proba(self, X: np.ndarray) -> np.ndarray:
        return self.model.predict_proba(X)[:, 1]

    def save(self, path: Path) -> None:
        joblib.dump(self.model, path)

    @classmethod
    def load(cls, path: Path) -> 'TransitClassifier':
        instance = cls()
        instance.model = joblib.load(path)
        return instance
```

### Training Script
```python
# src/models/train.py
def train_model(config_path: str = 'configs/model_config.yaml'):
    """Train transit classifier."""
    X_train, y_train = load_training_data()

    clf = TransitClassifier()
    clf.fit(X_train, y_train)
    clf.save(Path('models/rf_baseline.pkl'))
```

## Verification

```bash
python src/models/train.py --config configs/model_config.yaml
python src/models/evaluate.py --model models/rf_baseline.pkl
# Expected output: AUC: 0.87, Precision: 0.72, Recall: 0.65
```

## Definition of Done

- [ ] Model achieves >0.85 AUC on held-out test set
- [ ] Precision >0.7 at 0.5 recall threshold
- [ ] Confusion matrix and ROC curve saved
- [ ] Feature importance visualization created
- [ ] Model checkpoint saved and loadable

## Labels

`phase-3`, `modeling`, `priority-high`
